progress-emitter.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const EventEmitter = require('events').EventEmitter;
+
+class ProgressEmitter extends EventEmitter {
+	constructor(src, dest) {
+		super();
+		this.src = src;
+		this.dest = dest;
+
+		let written;
+		Object.defineProperty(this, 'written', {
```

Comment:
I would keep the private state in a WeakMap and make the `written` a prototype getter/setter. See: http://www.2ality.com/2016/01/private-data-classes.html